src/main.cpp

```diff
@@ -7181,9 +7181,9 @@ bool ProcessMessages(CNode *pfrom)
 #ifdef BITCOIN_CASH
         // This is a new peer. Before doing anything, we need to detect what magic the peer is using.
         if (pfrom->nVersion == 0 &&
-            memcmp(msg.hdr.pchMessageStart, chainparams.CashMessageStart(), MESSAGE_START_SIZE) == 0)
+            memcmp(msg.hdr.pchMessageStart, chainparams.MessageStart(), MESSAGE_START_SIZE) == 0)
         {
-            pfrom->fUsesCashMagic = true;
+            pfrom->fUsesCashMagic = false;
         }
 #endif
```

Comment:
I don't see why this whole section is still necessary.  If we're running the CASH version then the bits are already set, if not then the bits not set...whey do we need to change anything?  The way I read the code , if we send our VERSION message first then we stop using the cash magic...which will result in us getting disconnected later.  I believe, this section is always assuming that we first receive the VERSION message from a CASH node, which is not true, it can be either/or.

Reply:
If you don't do this, you'll ban all the older BUCash nodes.

Reply:
@ptschip in the last commit I made I change the default behavior when establishing a new outgoing conn.
new default is to use the cash magic, so if we receive a net msg using the legacy magic set we *before sending our version cmd* we need to switch back. 

I guess I could have made this change on a separate PR.